tests/e2e/specs/add-paid-campaigns/step-1-accounts.test.js

```diff
@@ -0,0 +1,254 @@
+ */
+let dashboardPage = null;
+
+/**
+ * @type {import('../../utils/pages/setup-ads/setup-ads-accounts').default} setupAdsAccounts
+ */
+let setupAdsAccounts = null;
+
+/**
+ * @type {import('@playwright/test').Locator} adsConnectionButton
+ */
+let adsConnectionButton = null;
+
+/**
+ * @type {import('@playwright/test').Page} page
+ */
+let page = null;
+
+test.describe( 'Set up Ads account', () => {
+	test.beforeAll( async ( { browser } ) => {
+		page = await browser.newPage();
+		dashboardPage = new DashboardPage( page );
+		setupAdsAccounts = new SetupAdsAccountsPage( page );
+		await setOnboardedMerchant();
+		await setupAdsAccounts.mockAdsAccountsResponse( [] );
+		await dashboardPage.mockRequests();
+		await dashboardPage.goto();
+	} );
+
+	test.afterAll( async () => {
+		await clearOnboardedMerchant();
+		await page.close();
+	} );
+
+	test( 'Dashboard page contains Add Paid campaign buttons', async () => {
+		//Add page campaign in the Performance (Paid Campaigns) section
+		await expect(
+			await dashboardPage.getAdsConnectionAllProgramsButton(
+				'summary-card'
+			)
+		).toBeEnabled();
+
+		//Add page campaign in the programs section.
+		adsConnectionButton =
+			await dashboardPage.getAdsConnectionAllProgramsButton();
+		await expect( adsConnectionButton ).toBeEnabled();
+	} );
+
+	test.describe( 'Set up your accounts page', async () => {
+		test.beforeAll( async () => {
+			await setupAdsAccounts.mockAdsAccountsResponse( [] );
+			await adsConnectionButton.click();
+			await page.waitForLoadState( LOAD_STATE.DOM_CONTENT_LOADED );
+		} );
+
+		test( 'Page header should be "Set up your accounts"', async () => {
+			await expect(
+				page.getByRole( 'heading', { name: 'Set up your accounts' } )
+			).toBeVisible();
+			await expect(
+				page.getByText(
+					'Connect your Google account and your Google Ads account to set up a paid Performance Max campaign.'
+				)
+			).toBeVisible();
+		} );
+
+		test( 'Google Account should show as connected', async () => {
+			await expect(
+				page.getByText(
+					'This Google account is connected to your store’s product feed.'
+				)
+			).toBeVisible();
+		} );
+
+		test( 'Continue Button should be disabled', async () => {
+			await expect(
+				await setupAdsAccounts.getContinueButton()
+			).toBeDisabled();
+		} );
+	} );
+
+	test.describe( 'Add paid campaigns with no Ads account', async () => {
+		test( 'Create an account should be visible', async () => {
+			const createAccountButton = page.getByRole( 'button', {
+				name: 'Create account',
+			} );
+
+			await expect( createAccountButton ).toBeVisible();
+
+			await expect(
+				await setupAdsAccounts.getContinueButton()
+			).toBeDisabled();
+
+			await expect(
+				page.getByText(
+					'Connect with millions of shoppers who are searching for products like yours and drive sales with Google.'
+				)
+			).toBeVisible();
+
+			await createAccountButton.click();
```

Comment:
What do you think if we make the _**clicking create account button**_ as a separate test so it would be a bit more descriptive? The tests below could potentially be separated as

- Click create account button
- Click accept ToS checkbox
- Click create ads account button in the modal

Reply:
Yeah I think it is a good idea, I addressed here: https://github.com/woocommerce/google-listings-and-ads/pull/2097/commits/c4691d015255649b5a57bfe39753564963c5a8ac